.github/workflows/validate.yml

```diff
@@ -0,0 +1,29 @@
+name: validate
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check for sources
+        run: |
+          filecount=`find . -name '*.dart' | wc -l`
```

Comment:
Should we just check in `lib/`?

I imagine it would be valuable to have some dart code in `tool/` for instance, in order to do something like validate that the analysis options file is well-formed.

Reply:
Then we should check `bin` too

Reply:
Ah.  That's a good call.  In that case, maybe we just fail if there's a `lib` dir and this can get a bit simpler?

Reply:
@mit-mit: right.  I'll check for both.

Reply:
Checking bin/ and lib/ makes sense.

> maybe we just fail if there's a lib dir and this can get a bit simpler?

Well, we're definitely need lib/analysis_options.yaml (or lib/score.yaml, ...) :)

Reply:
Of course.  Right.   Will fix up.